test/memory.test.js

```diff
@@ -76,6 +85,35 @@ describe('Memory connector', function() {
       }, done);
     });
 
+    it('should not have out of sequence read/write', function(done) {
```

Comment:
I'm using a side effect here to make sure existing records in the file (produced by `create`) won't be overridden.